docs/spark-connect-overview.md

```diff
@@ -56,6 +56,26 @@ client through gRPC as Apache Arrow-encoded row batches.
   <img src="img/spark-connect-communication.png" title="Spark Connect communication" alt="Spark Connect communication" />
 </p>
 
+## What is changing with Spark Connect
```

Comment:
This heading makes it sound like Spark Connect used to work differently and is now changing. That is not the case, is it?

I think a better heading would be something along the lines of: "How Spark Connect differs from "traditional" Spark programs"

I couldn't find a better word to describe non-Connect programs than "traditional", but you get the gist.